samples/go/client/client.go

```diff
@@ -0,0 +1,146 @@
+package client
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+
+	"a2a/models"
+)
+
+// Client represents an A2A protocol client
+type Client struct {
+	baseURL    string
+	httpClient *http.Client
+}
+
+// NewClient creates a new A2A client
+func NewClient(baseURL string) *Client {
+	return &Client{
+		baseURL: baseURL,
+		httpClient: &http.Client{
+			Timeout: 30 * time.Second,
+		},
+	}
+}
+
+// SendTask sends a task message to the agent
```

Comment:
I don't see streaming support, is this something you plan on contributing in the future? Here is our Python reference implementation equivalent: https://github.com/google/A2A/blob/main/samples/python/common/client/client.py#L38